include/mxnet/ndarray.h

```diff
@@ -102,15 +112,29 @@ class NDArray {
   NDArray(const NDArrayStorageType stype, const mxnet::TShape &shape, Context ctx,
           bool delay_alloc = true, int dtype = mshadow::default_type_flag,
           std::vector<int> aux_types = {}, mxnet::ShapeVector aux_shapes = {},
-          mxnet::TShape storage_shape = mxnet::TShape(mshadow::Shape1(0)));
+          mxnet::TShape storage_shape = mxnet::TShape(mshadow::Shape1(0))
+#if MXNET_ENABLE_STORAGE_TAGGING
```

Comment:
Why not always have this enabled and not have the define? memory concerns?

Reply:
Mostly performance concerns. Since the GPU memory profiler touches both the Python frontend and the C++ backend, it is hard to measure the actual overhead of propagating the name tags all the way down from the Python APIs to the C++ storage handles. Hence, this flag is added for people who just want to work with a "clean" version of MXNet.